dotCMS/src/main/java/com/dotcms/cluster/business/ServerAPIImpl.java

```diff
@@ -35,52 +44,111 @@ public void saveServer(Server server) throws DotDataException {
+                Config.setProperty(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName(), port);
+                newServer.setEsTransportTcpPort(Integer.parseInt(port));
+    
+                port=ClusterFactory.getNextAvailablePort(newServer.getServerId(), ServerPort.ES_HTTP_PORT);
+                Config.setProperty(ServerPort.ES_HTTP_PORT.getPropertyName(), port);
+                newServer.setEsHttpPort(Integer.parseInt(port));
+    
+               updateServer(newServer);
+    
+                try {
+                    writeHeartBeatToDisk(newServer.getServerId());
+                } catch (IOException e) {
+                    Logger.error(ClusterFactory.class, "Could not write Server ID to file system" , e);
+                }
+       
+            });
 
-	public String readServerId() {
-
-		String realPath = ConfigUtils.getDynamicContentPath() + java.io.File.separator + "server_id.dat";
-
-		Logger.debug(ServerAPIImpl.class, "Reading " + realPath);
-
-		BufferedReader br = null;
-		String serverId = null;
-		try {
-			br = new BufferedReader(new FileReader(new File(realPath)));
-			serverId = br.readLine();
-		} catch (FileNotFoundException e) {
-			Logger.debug(ServerAPIImpl.class, "Server ID not found (" + realPath + " file doesn't exists)");
-		} catch (IOException e) {
-			Logger.error(ServerAPIImpl.class, "Could not read Server ID from File", e);
-		} finally {
-			try {
-				if(br!=null)
-					br.close();
-			} catch (IOException e) {
-				Logger.error(ServerAPIImpl.class, "Could not close BufferedReader for Server File", e);
-			}
-		}
-
-		Logger.debug(ServerAPIImpl.class, "ServerID: " + serverId);
-
-        return serverId;
-
-	}
+        }
+        
 
-	public  void writeServerIdToDisk(String serverId) throws IOException {
+        return serverFactory.getServer(serverId);
+    }
 
+	private File serverIdFile(){
+	    
         String dynamicContentPath = ConfigUtils.getDynamicContentPath();
 
+        
         String realPath;
         if ( dynamicContentPath.endsWith( File.separator ) ) {
             realPath = ConfigUtils.getDynamicContentPath() + "server_id.dat";
         } else {
             realPath = ConfigUtils.getDynamicContentPath() + File.separator + "server_id.dat";
         }
-        File serverFile = new File(realPath);
+        Logger.debug(ServerAPIImpl.class, "Server Id " + realPath);
 
-		if(serverFile.exists())
-			serverFile.delete();
+        return new File(realPath);
 
+	}
+	
+	
+	
+	
+	@Override
+	public String readServerId()  {
+	    // once set this should never change
+	    if(SERVER_ID==null){
+    	    try{
+        	    File serverFile = serverIdFile();
+        
+        	    if(!serverFile.exists()){
+        	        writeServerIdToDisk(UUIDUtil.uuid());
+        	    }
+        
+        		BufferedReader br =  new BufferedReader(new FileReader(serverFile));
+        		SERVER_ID = br.readLine();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'SERVER_ID' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2444)